### PR TITLE
Resolve overlapping resource access between user and group

### PR DESCRIPTION
**Pull request by pkdash**
This PR address #1241

### CODE REVIEW

**Comment by pkdash** [comment]
@hyi Can you review this? This PR needs to be merged first before I can create a PR for #1242

**Comment by hydrobuild** [comment]
Can one of the admins verify this patch?

**Comment by alvacouch** [comment]
@pkdash While these changes are not problematic from semantic perspective, the intent does not make much sense to me yet. 

This doesn't seem to change the case in which someone is granted personal access and then group access .... and it shouldn't change that. All it changes is whether the person can be granted personal access if already a member of a group with access. 

I don't really understand why this case should be prevented. 

Perhaps let's talk about this? I am at the data center: 339 221 5400 x203.

**Comment by alvacouch** [comment]
It strikes me that there are many other places that "HydroShare Author" should not be returned as a group result.

File: hs_core/autocomplete_light_registry.py [hunk 1]
```diff
@@ -37,7 +37,7 @@ class GroupAutocomplete(autocomplete_light.AutocompleteModelBase):
     search_fields=['name']
 
     def choices_for_request(self):
-        self.choices = self.choices.filter(gaccess__active=True)
+        self.choices = self.choices.filter(gaccess__active=True).exclude(name='Hydroshare Author')
```
Comment (alvacouch) [inline hs_core/autocomplete_light_registry.py]: We need this fix system wide. The fact that this group is returned is an oversight.

File: hs_core/views/__init__.py [hunk 2]
```diff
@@ -696,7 +696,7 @@ class AddUserForm(forms.Form):
                                       widget=autocomplete_light.ChoiceWidget("UserAutocomplete"))
 
     class AddGroupForm(forms.Form):
-        group = forms.ModelChoiceField(Group.objects.filter(gaccess__active=True).all(),
+        group = forms.ModelChoiceField(Group.objects.filter(gaccess__active=True).exclude(name='Hydroshare Author').all(),
```
Comment (alvacouch) [inline hs_core/views/__init__.py]: I think there shouldn't be a gaccess record for "HydroShare Author".... That would fix this system wide.

**Comment by alvacouch** [comment]
@pkdash @hyi Based upon my analysis of the code I am willing to +1 this. Hong, do you agree?

**Comment by alvacouch** [comment]
Talked with @pkdash; resolved my concerns.

File: hs_core/views/__init__.py (hunk 2 above)
Comment (hyi) [inline hs_core/views/__init__.py]: Agreed, but it is good to filter out this HydroShare Author group just to be sure for now.

**Comment by hyi** [comment]
+1 @pkdash @alvacouch

**Pull request merged**